### PR TITLE
API Integrations

### LINKED ISSUE #32: (0,y.auth0Logout) is not a function. (In '(0,y.auth0Logout)()', '(0,y.auth0Logout)' is undefined)
## Error in Tree Mapper

**TypeError** in **main.jsbundle:1176**
(0,y.auth0Logout) is not a function. (In '(0,y.auth0Logout)()', '(0,y.auth0Logout)' is undefined)

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/tree-mapper/errors/5f2139430714aa00178225a5?event_id=5f213943005b71959cc00000&i=gh&m=ci)

## Stacktrace

    main.jsbundle:1176 - onPressLogout

[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/tree-mapper/errors/5f2139430714aa00178225a5?event_id=5f213943005b71959cc00000&i=gh&m=ci)

*Created automatically via Bugsnag*

### PR DESCRIPTION
**Pull request by sagararyal**
Fixes #32 

Changes in this pull request:
- Login/ Signup with Auth0
- Flow Improvement & Fixes
- Integration  of Upload API
-
-

@

### CODE REVIEW

File: app/Components/RegisterSingleTree/TreeOverview.js [hunk 1]
```diff
@@ -77,7 +79,7 @@ const SingleTreeOverview = ({ navigation }) => {
                             style={styles.bgWhite}>
                             <View style={styles.externalInputContainer}>
                                 <Text style={styles.labelModal}>{isSpeciesEnable ? 'Name of Specie' : 'Diameter'}</Text>
-                                {isSpeciesEnable ? <TextInput value={specieText} style={styles.value} autoFocus placeholderTextColor={Colors.TEXT_COLOR} onChangeText={(text) => setSpecieText(text)} onSubmitEditing={() => onSubmitInputFeild('specieText')} keyboardType={'default'} />
+                                {isSpeciesEnable ? <TextInput value={specieText} style={styles.value} autoFocus placeholderTextColor={Colors.TEXT_COLOR} onChangeText={(text) => setSpecieText(text)} onSubmitEditing={() => onSubmitInputFeild('specieText')} keyboardType={'email-address'} />
                                     : <TextInput ref={specieDiameterRef} value={specieDiameter} style={styles.value} autoFocus placeholderTextColor={Colors.TEXT_COLOR} keyboardType={'number-pad'} onChangeText={(text) => setSpecieDiameter(text)} onSubmitEditing={() => onSubmitInputFeild('specieDiameter')} />}
```
Comment (sagararyal) [inline app/Components/RegisterSingleTree/TreeOverview.js:83]: Change label of `Diameter` to `Diameter (in cm)`

File: app/Components/RegisterSingleTree/TreeOverview.js [hunk 2]
```diff
@@ -105,39 +107,38 @@ const SingleTreeOverview = ({ navigation }) => {
...
-                <View>
-                    <Text style={detailHeaderStyle}>Planting Date</Text>
-                    <TouchableOpacity disabled={!shouldEdit} onPress={() => setIsShowDate(true)}>
-                        <Text style={styles.detailText}>{new Date(plantationDate).toLocaleDateString()} {shouldEdit && <MIcon name={'edit'} size={20} />}</Text>
-                    </TouchableOpacity>
-                </View>
+            <View style={detailContainerStyle}>
                 <View>
                     <Text style={detailHeaderStyle}>LOCATION</Text>
                     <Text style={styles.detailText}>{`${coords.latitude.toFixed(5)}˚N,${coords.longitude.toFixed(5)}˚E`} </Text>
```
Comment (sagararyal) [inline app/Components/RegisterSingleTree/TreeOverview.js:144]: This calculation for N, E is incorrect
If decimals are used, N , E is not necessary.

this needs to be converted to sexadecimal; -10 = 10 degree S;
or -10 = 10 degree West

File: app/Components/RegisterSingleTree/TreeOverview.js [hunk 3]
```diff
@@ -166,14 +167,28 @@ const SingleTreeOverview = ({ navigation }) => {
...
             navigation.navigate('TreeInventory')
-        })
+        } else {
+            if (specieText) {
+                let data = { inventory_id: state.inventoryID }
+                statusToPending(data).then(() => {
+                    navigation.navigate('TreeInventory')
+                })
+            } else {
+                alert('Specie Name  is required')
```
Comment (sagararyal) [inline app/Components/RegisterSingleTree/TreeOverview.js:189]: Species (is both singular and plural)

File: app/Components/SignUp/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                        <View style={[[styles.roleBtnContainer, styles.activeRoleContainer], styles.justifyCenter, styles.marginRight]}>
+                            <Text style={[styles.roleText, styles.primaryText]}>Tree Planting Organisation</Text>
+                        </View>
+                        <View style={[styles.roleBtnContainer, styles.marginLeft]}>
+                            <Text style={styles.roleText}>School</Text>
+                        </View>
+                    </View>
+                    <Input label={'NAME OF TREE PLANTING ORGANIZATION'} value={'Forest in Africa'} />
+                    <View style={styles.switchContainer}>
+                        <Text style={styles.switchContainerText}>I agree to have my name published on the Plant-for-the-Planet App.</Text>
```
Comment (sagararyal) [inline app/Components/SignUp/index.js:37]: this is `mayPublish` flag in API

File: app/Components/SignUp/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                        <Switch trackColor={{ false: Colors.LIGHT_BORDER_COLOR, true: Colors.PRIMARY }} thumbColor={!false ? Colors.PRIMARY : Colors.WHITE} value={!false} />
+                    </View>
+                    <View style={styles.switchContainer}>
+                        <Text style={styles.switchContainerText}>I agree that I may be contacted by Plant-for-the-Planet as a part of Tree Planting news and challenges.</Text>
```
Comment (sagararyal) [inline app/Components/SignUp/index.js:41]: this  is `mayContact` flag in API

File: app/Components/TreeInventory/index.js [hunk 6]
```diff
@@ -62,18 +85,43 @@ const TreeInventory = ({ navigation }) => {
...
+        })
+    }
+
+    const renderLoaderModal = () => {
+        return (<Modal
+            transparent
+            visible={isLoaderShow}>
+            <View style={styles.dowloadModalContainer}>
+                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+                    <Text style={{ padding: 30, backgroundColor: '#fff', borderRadius: 10 }}>Uploading .......</Text>
```
Comment (sagararyal) [inline app/Components/TreeInventory/index.js:111]: Before showing this modal, add a check if user is logged In.
If user is not logged in, redirect to login screen; log them in; before uploading.

File: ios/treemapper/Info.plist [hunk 7]
```diff
@@ -38,13 +38,13 @@
 		</dict>
 	</dict>
 	<key>NSCameraUsageDescription</key>
-	<string>Allow access to your camera</string>
+	<string>To take pictures of the trees you have planted, please allow access to your camera.</string>
 	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
```
Comment (sagararyal) [inline ios/treemapper/Info.plist:42]: We do not need access when the app is not being used. So need to disable that.

File: app/Actions/index.js [hunk 8]
```diff
@@ -258,7 +301,8 @@ export const addCoordinateSingleRegisterTree = ({ inventory_id, markedCoords, cu
                         }]
                     }]
                     inventory.specei_diameter = 10
```
[the remark this answers is not in the document]
Comment (haideralishah) [reply app/Actions/index.js]: fixed

File: app/Components/RegisterSingleTree/TreeOverview.js (hunk 1 above)
Comment (haideralishah) [reply app/Components/RegisterSingleTree/TreeOverview.js:83]: fixed

File: app/Components/RegisterSingleTree/TreeOverview.js (hunk 3 above)
Comment (haideralishah) [reply app/Components/RegisterSingleTree/TreeOverview.js]: Fixed

File: app/Components/TreeInventory/index.js (hunk 6 above)
Comment (haideralishah) [reply app/Components/TreeInventory/index.js:132]: Fixed

**Pull request merged**